bears/vcs/git/GitCommitBear.py

```diff
@@ -151,6 +161,53 @@ def run(self,
             body,
             **self.get_issue_checks_metadata().filter_parameters(kwargs))
 
+    def check_revert(self, shortlog, body,
+                     allow_revert_commits_without_reason: bool = False,):
+        """
+        Validates revert commits.
+
+        :param shortlog:                 The shortlog message string.
+        :param body:          The body of the commit message of HEAD.
+        :param allow_revert_commits_without_reason: Whether reason is
+                                           needed for revert commits.
+        """
+        revert_shortlog_regex = re.compile(r'^Revert\s"([\S+\s*]*)"$')
+        headline_regex = re.compile(r'^This reverts commit [0-9a-f]{40}')
+        sha_regex = re.compile(r'\b[0-9a-f]{40}\b')
+
+        if (not allow_revert_commits_without_reason and
```

Comment:
Question:
What will happen in case the user allows revert commits without reason and while generating a revert commit, say the shortlog doesn't match or the commit sha is altered for some reason ? (via a ```git commit --amend``` for example)